src/pkg/cluster/injector.go

```diff
@@ -263,6 +263,9 @@ func (c *Cluster) getInjectorImageAndNode(ctx context.Context, resReq *v1ac.Reso
 		if err != nil {
 			return "", "", err
 		}
+		if nodeDetails.Status.NodeInfo.Architecture == "" || nodeDetails.Status.NodeInfo.Architecture != architecture {
+			continue
+		}
```

Comment:
```suggestion
		if nodeDetails.Status.NodeInfo.Architecture != "" && nodeDetails.Status.NodeInfo.Architecture != architecture {
			continue
		}
```
Suggestion to change the behavior to not continue if architecture is empty. That way on the off chance there's some distro or version that doesn't fill in this value, nothing breaks for existing users

Reply:
I will have to review, but I think that there was an issue with the injector logic not finding the right node with the suggested config.... but I will review and see if I had some bad logic before